examples/basic.py

```diff
@@ -22,13 +22,12 @@ def peek(x):
     print(f"peekin at {x}")
 
 
-ec = bytewax.Executor()
-flow = ec.Dataflow(inp())
+flow = Dataflow()
 flow.map(double)
 flow.map(minus_one)
 flow.map(stringy)
-flow.inspect(peek)
+flow.inspect(print)
 
 
 if __name__ == "__main__":
-    ec.build_and_run()
+    run_cluster(flow, inp(), **parse.cluster_args())
```

Comment:
Huh. Why did the tests not catch this? I thought `enumerate` is a generator. Let me do some more investigation. I have not been running all of the examples before PRing this since I wanted to get feedback on the shape of things first and also because I am lazy. (But this makes me all the more want to get something like doctests working in our CI so we can make sure all the examples are working automatically.) I also need to think through if this needs to support "infinite input."

Reply:
Ok. I guess there's some sort of trick where enumerate can be pickled. To fix this, I'll cast to a list to turn the whole generator into a list. This means you need to be able to fit the whole input in memory, but as of now `run_sync` and `run_cluster` can't really support infinite input anyway because they store all the output in a list (which would grow without bound). You can use the `main_*` functions with `output_builder` to handle those cases.

In the future, I'd like to figure out how to make a `run_async`and `run_cluster_async` which support streaming output, but I think that'll require some async PyO3 coroutine bridge stuff.